src/sentry/static/sentry/app/components/events/eventEntries.tsx

```diff
@@ -45,39 +65,174 @@ type Props = {
+    if (!str) {
+      return false;
+    }
+
+    return !![...str.matchAll(MATCH_MINIFIED_DATA_REGEX)].length;
+  }
+
+  hasThreadOrExceptionMinifiedFrameData(event: Event, bestThread?: Thread) {
+    if (!bestThread) {
+      const exceptionValues: Array<ExceptionValue> =
+        event.entries?.find(e => e.type === EntryType.EXCEPTION)?.data?.values ?? [];
+
+      return !!exceptionValues.find(exceptionValue =>
+        exceptionValue.stacktrace?.frames?.find(frame =>
+          this.isDataMinified(frame.module)
+        )
+      );
+    }
+
+    const threadExceptionValues = getThreadException(event, bestThread)?.values;
+
+    return !!(threadExceptionValues
+      ? threadExceptionValues.find(threadExceptionValue =>
+          threadExceptionValue.stacktrace?.frames?.find(frame =>
+            this.isDataMinified(frame.module)
+          )
+        )
+      : bestThread?.stacktrace?.frames?.find(frame => this.isDataMinified(frame.module)));
+  }
+
+  async checkProGuardError() {
     const {event} = this.props;
 
-    if (!event || !event.errors || !(event.errors.length > 0)) {
+    if (!event) {
       return;
     }
-    const errors = event.errors;
-    const errorTypes = errors.map(errorEntries => errorEntries.type);
-    const errorMessages = errors.map(errorEntries => errorEntries.message);
 
-    this.recordIssueError(errorTypes, errorMessages);
-  }
+    const hasEventErrorsProGuardMissingMapping = event.errors?.find(
+      error => error.type === 'proguard_missing_mapping'
+    );
 
-  shouldComponentUpdate(nextProps: Props) {
-    const {event, showExampleCommit} = this.props;
+    if (hasEventErrorsProGuardMissingMapping) {
+      this.setState({isLoading: false});
+      return;
+    }
 
-    return (
-      (event && nextProps.event && event.id !== nextProps.event.id) ||
-      showExampleCommit !== nextProps.showExampleCommit
+    const proGuardErrors: ProGuardErrors = [];
+
+    const debugImages = event.entries?.find(e => e.type === EntryType.DEBUGMETA)?.data
+      .images as undefined | Array<Image>;
+
+    // When debugImages contains a 'proguard' entry, it must always be only one entry
+    const proGuardImage = debugImages?.find(
+      debugImage => debugImage?.type === 'proguard' && !!debugImage?.uuid
     );
+
+    // If an entry is of type 'proguard' and has 'uuid',
+    // it means that the Sentry Gradle plugin has been executed,
+    // otherwise the proguard id wouldn't be in the event.
+    // But maybe it failed to upload the mappings file
+    if (proGuardImage?.uuid) {
+      const debugFile = await this.fetchDebugFile(proGuardImage.uuid);
+
+      if (!debugFile.length) {
+        proGuardErrors.push({
+          type: 'proguard_missing_mapping',
+          message: projectProcessingIssuesMessages.proguard_missing_mapping,
+          data: {mapping_uuid: proGuardImage.uuid},
+        });
+      }
+
+      this.setState({proGuardErrors, isLoading: false});
+      return;
+    }
```

Comment:
If we had the `proguard` node but not `uuid`, can we raise a Sentry event for us to investigate? I'm assuming the Sentry event would link to this event so we can get back to it.

Ideally we would tag the SDK version, pseudo code like:

```javascript
Sentry.withScope(function(s) {
  s.setLevel(Sentry.Severity.Warning);
  s.setTag("offending.event.sdk.name", event.sdk.name);
  s.setTag("offending.event.sdk.version", event.sdk.version);
  Sentry.captureMessage("Event contains proguard but not uuid");
});
```